README.md

```diff
@@ -27,6 +33,20 @@ If you're one of the brave few who wants to hack the firmware then read on.
 
 Going forward, it's easier to flash the firmware of your choice by using the downwards toolbar icon which is located rightwards of the *green play + toolbox icon*.
 
+### Minimal development setup
+
+1. Install the ARM cross-compiler, cross-assembler and stdlib implementation. Eg. on Arch Linux the packages `arm-none-eabi-binutils`, `arm-none-eabi-gcc`, `arm-none-eabi-newlib`.
+
+2. Install Node.js v12. If you have a later version, editing the version requirement in `lib/agent/package.json` *might* work.
+
+3. Build UHK Agent. `cd lib/agent && git pull --autostash && npm ci && npm run build`.
+
+4. Still inside the Agent submodule, compile flashing util scripts. `cd packages/usb && npx tsc`.
```

Comment:
`npm run build` should take care of this.

Reply:
It didn't for me. Maybe some editor plugin that you're using automatically transpiles typescript to javascript?

Reply:
@ert78gb Can you comment on this?

Reply:
the `npm run build` compile the bundles from the ts files. It doesn't compile the ts files in `lib/agent/packages/usb`, but he step 4 do it.

@dancek if you experienced the bundles not compiled for you please check the output of the build command. If you don't find the error inside then send us the output for me and I will help.

The ts files in the `lib/agent/packages/usb` starts with the `#!/usr/bin/env ../../node_modules/.bin/ts-node-script` shebang so it will compile at runtime. If someone use it frequently and use low capacity cpu-s then worth to compile them with the step 4.

Reply:
Agent can be develop without any IDE.

Reply:
The flashing scripts in this repo use the compiled js files. Eg. https://github.com/UltimateHackingKeyboard/firmware/blob/master/right/Makefile#L32

```
FLASH_CMD = node ../lib/agent/packages/usb/update-device-firmware.js $(PROJECT_OBJ:.axf=.hex)
```

This is why I added step 4. I do realize there are other ways, but I wanted to document how this works now rather than change it.

Reply:
Recently, @ert78gb converted the js files to ts, so the latter should be used, and Makefiles should be updated accordingly.